roles/ansible-core-ci-session/tasks/main.yaml

```diff
@@ -0,0 +1,16 @@
+---
+- name: Ensure create_core_ci_session_key is set
+  assert:
+    that:
+      - create_core_ci_session_key is defined
+      - create_core_ci_session_key|length > 0
+- name: Create a new session
+  uri:
+    method: PUT
+    body_format: json
+    body: '{"config": {"platform": "aws", "version": "sts"}, "auth": {"remote": {"key": "{{ create_core_ci_session_key }}", "nonce": null}}, "threshold": 1}'
+    url: "https://ansible-core-ci.testing.ansible.com/{{ create_core_ci_session_stage }}/aws/{{ create_core_ci_instance_id }}"
+    dest: "{{ create_core_ci_local_target_file }}"
+    mode: '0400'
+  no_log: true
+  register: my_session
```

Comment:
registered variables do not persist across plays. What we should do, is write the session to disk, so the next level of the job can access it.

Reply:
this is what I do, see line 13.

Reply:
ack, I see it now. We could delete my_session here, as it would never be used.

Reply:
Oh indeed. I understand your confusion now.